packages/devtools/lib/src/framework/framework.dart

```diff
@@ -83,50 +83,49 @@ class Framework {
 
     final droppedFile = files.first;
     if (droppedFile.type != 'application/json') {
-      toast('${droppedFile.type} is not a supported file type.\n Please import '
+      toast('${droppedFile.type} is not a supported file type. Please import '
           'a .json file that was exported from Dart DevTools.');
       return;
     }
 
     final FileReader reader = FileReader();
     reader.onLoad.listen((_) {
-      final Map<String, dynamic> import = jsonDecode(reader.result);
-      final devToolsScreen = import['dartDevToolsScreen'];
+      try {
+        final Map<String, dynamic> import = jsonDecode(reader.result);
+        final devToolsScreen = import['dartDevToolsScreen'];
+
+        if (devToolsScreen == null) {
+          toast(
+            'The imported file is not a Dart DevTools file. At this time, '
+            'DevTools only supports importing files that were originally '
+            'exported from DevTools.',
+            hideDelay: Toast.extendedHideDelay,
+          );
+          return;
+        }
 
-      if (devToolsScreen == null) {
+        // TODO(jacobr): add the inspector handling case here once the inspector
+        // can be exported.
+        switch (devToolsScreen) {
+          case timelineScreenId:
+            _importTimeline(import);
+            break;
+          default:
+            toast(
+              'Could not import file. The imported file is from an unrecognized'
+              ' DevTools screen: $devToolsScreen.',
+              hideDelay: Toast.extendedHideDelay,
+            );
+        }
+      } on FormatException catch (e) {
         toast(
-          'The imported file is not a Dart DevTools file. At this time, '
-          'DevTools only supports importing files that were originally exported'
-          ' from DevTools.',
+          'SyntaxError in imported file: "$e". Please make sure the imported '
```

Comment:
SyntaxError in --> JSON syntax error in
?

Reply:
done,.